sdk/test.go

```diff
@@ -175,30 +175,33 @@ func doDial(endpoint string, insecure bool, authProtocolType string, authToken s
 		return nil, err
 	}
 
-	if res.StatusCode != 307 {
-		return nil, customError(res)
-	}
-
-	log.Debug("* Received 307 redirect")
-
-	location, err := res.Location()
-	if err != nil {
-		log.Error("could not get location off header")
-		return nil, err
-	}
+	switch res.StatusCode {
+	case 307:
+		log.Debug("* Received 307 redirect")
+
+		location, err := res.Location()
+		if err != nil {
+			log.Error("could not get location off header")
+			return nil, err
+		}
 
-	conn, res, err = websocketDial(location.String(), insecure, authProtocolType, authToken)
-	if err != nil {
-		if res != nil {
-			customErr := customError(res)
-			res.Body.Close()
-			return nil, customErr
+		conn, res, err = websocketDial(location.String(), insecure, authProtocolType, authToken)
+		if err != nil {
+			if res != nil {
+				customErr := customError(res)
+				res.Body.Close()
+				return nil, customErr
+			}
+			log.Error("could not dial websocket again after 307 redirect")
+			return nil, err
 		}
-		log.Error("could not dial websocket again after 307 redirect")
-		return nil, err
-	}
 
-	return conn, nil
+		return conn, nil
+	case 404:
+		return nil, fmt.Errorf("code: %d, reason: could not establish connection to provided url", res.StatusCode)
```

Comment:
not sure this is the proper message for a 404 or here. We return 404 when it can't find the function? Perhaps we should say that?

Reply:
In this case, the error is because the url provided isn't valid because it doesn't exist or there are network issues, so maybe a more generic "could not connect to" message? No preference, whatever is the better user experience.